### PR TITLE
Add: document question answering task guide

### PR DESCRIPTION
**Pull request by MKhalusova**
This PR adds a task guide on document question answering. 

The example is largely based on @NielsRogge 's [notebook](https://github.com/NielsRogge/Transformers-Tutorials/blob/master/LayoutLMv2/DocVQA/Fine_tuning_LayoutLMv2ForQuestionAnswering_on_DocVQA.ipynb) with some inspuration from [this blog post](https://www.philschmid.de/fine-tuning-layoutlm#3-fine-tune-and-evaluate-layoutlm)

A Colab notebook to test the code examples is [here](https://colab.research.google.com/drive/1vIc-YoGbWdNOBGSb6mpkw9myJ4nr2vcA?usp=sharing).

### CODE REVIEW

File: docs/source/en/tasks/document_question_answering.mdx [hunk 1]
```diff
@@ -0,0 +1,485 @@
...
+a question, and the output is an answer expressed in natural language. These models utilize multiple features, including
+text, the positions of words (bounding boxes), and the image itself.
+
+This guide illustrates how to:
+
+- Fine-tune [LayoutLMv2](../model_doc/layoutlmv2) on the [DocVQA dataset](https://huggingface.co/datasets/nielsr/docvqa_1200_examples_donut).
+- Use your fine-tuned model for inference.
+
+<Tip>
+The task illustrated in this tutorial is supported by the following model architectures:
```
Comment (sgugger) [inline docs/source/en/tasks/document_question_answering.mdx:28]:
```suggestion
<Tip>

The task illustrated in this tutorial is supported by the following model architectures:
```

File: docs/source/en/tasks/document_question_answering.mdx [hunk 2]
```diff
@@ -0,0 +1,485 @@
...
+pip install -q transformers
+```
+
+```bash
+pip install pyyaml==5.1
+pip install torch==1.8.0+cu101 torchvision==0.9.0+cu101 -f https://download.pytorch.org/whl/torch_stable.html
+
+# install detectron2 that matches pytorch 1.8
+# See https://detectron2.readthedocs.io/tutorials/install.html for instructions
+pip install -q detectron2 -f https://dl.fbaipublicfiles.com/detectron2/wheels/cu101/torch1.8/index.html
```
Comment (sgugger) [inline docs/source/en/tasks/document_question_answering.mdx:55]: Is there a way to use a more recent version? Pytorch 1.8 is quite old.

File: docs/source/en/tasks/document_question_answering.mdx [hunk 3]
```diff
@@ -0,0 +1,485 @@
...
+let's remove the few examples where the embedding is likely to end up longer than 512. If most of the documents in your dataset are long, you
+may need to choose a different pretrained model checkpoint.
+
+```py
+>>> updated_dataset = updated_dataset.filter(lambda x: len(x["words"]) + len(x["question"].split()) < 512)
+```
+
+At this point let's also remove the OCR features from this dataset. These are a result of OCR for fine-tuning a different
+model. They would still require some processing if we wanted to use them, as they do not match the input requirements
+of the model we use in this guide. Instead, we can use the [`~LayoutLMv2Processor`] on the original data for both OCR and
```
Comment (sgugger) [inline docs/source/en/tasks/document_question_answering.mdx:144]:
```suggestion
of the model we use in this guide. Instead, we can use the [`LayoutLMv2Processor`] on the original data for both OCR and
```

File: docs/source/en/tasks/document_question_answering.mdx [hunk 4]
```diff
@@ -0,0 +1,485 @@
...
+## Preprocess the data
+
+The Document Question Answering task is a multimodal task, and you need to make sure that the inputs from each modality
+are preprocessed according to model's expectations. Let's start by loading the [`~LayoutLMv2Processor`], which internally
+combines an image processor that can handle image data and a tokenizer that can encode text data.
+
+```py
+>>> from transformers import LayoutLMv2Processor
+
+>>> processor = LayoutLMv2Processor.from_pretrained(model_checkpoint)
```
Comment (sgugger) [inline docs/source/en/tasks/document_question_answering.mdx:172]: It would be better to use AutoProcessor here, so this task guide can work with another architecture (as advertised at the top)

File: docs/source/en/tasks/document_question_answering.mdx [hunk 5]
```diff
@@ -0,0 +1,485 @@
...
+```
+
+### Preprocessing document images
+
+First, let's prepare the document images for the model with the help of the `image_processor` from the `LayoutLMv2Processor`.
```
Comment (sgugger) [inline docs/source/en/tasks/document_question_answering.mdx:177]:
```suggestion
First, let's prepare the document images for the model with the help of the `image_processor` from the processor.
```

File: docs/source/en/tasks/document_question_answering.mdx [hunk 6]
```diff
@@ -0,0 +1,485 @@
...
+
+```py
+>>> dataset_with_ocr = updated_dataset.map(get_ocr_words_and_boxes, batched=True, batch_size=2)
+```
+
+### Preprocessing text data
+
+Once we have applied OCR to the images, we need to encode the text part of the dataset to prepare it for the model.
+This involves converting the words and boxes that we got in the previous step to token-level `input_ids`, `attention_mask`,
+`token_type_ids` and `bbox`. For preprocessing text, we'll need the `tokenizer` from the `LayoutLMv2Processor`.
```
Comment (sgugger) [inline docs/source/en/tasks/document_question_answering.mdx:207]:
```suggestion
`token_type_ids` and `bbox`. For preprocessing text, we'll need the `tokenizer` from the processor.
```

File: docs/source/en/tasks/document_question_answering.mdx [hunk 7]
```diff
@@ -0,0 +1,485 @@
...
+* Load the model with [`AutoModelForQuestionAnswering`] using the same checkpoint as in the preprocessing.
+* Define your training hyperparameters in [`TrainingArguments`].
+* Define a function to batch examples together, here the [`DefaultDataCollator`] will do just fine
+* Pass the training arguments to [`Trainer`] along with the model, dataset, and data collator.
+* Call [`~Trainer.train`] to finetune your model.
+
+```py
+>>> from transformers import AutoModelForQuestionAnswering
+
+>>> model = AutoModelForQuestionAnswering.from_pretrained(model_checkpoint)
```
Comment (sgugger) [inline docs/source/en/tasks/document_question_answering.mdx:372]: Shouldn't this be the `ForDocumentQuestionAnswering` class here?

File: docs/source/en/tasks/document_question_answering.mdx [hunk 8]
```diff
@@ -0,0 +1,485 @@
...
+which token is at the end of the answer. Both have shape (batch_size, sequence_length).
+4. Take an argmax on the last dimension of both the `start_logits` and `end_logits` to get the predicted `start_idx` and `end_idx`.
+5. Decode the answer with the tokenizer.
+
+```py
+>>> import torch
+>>> from transformers import LayoutLMv2Processor
+>>> from transformers import AutoModelForDocumentQuestionAnswering
+
+>>> processor = LayoutLMv2Processor.from_pretrained("MariaK/layoutlmv2-base-uncased_finetuned_docvqa")
```
Comment (sgugger) [inline docs/source/en/tasks/document_question_answering.mdx:472]: Should use AutoProcessor here.

**Review by sgugger — COMMENTED** [review]
Thanks for working on this!

File: docs/source/en/tasks/document_question_answering.mdx (hunk 2 above)
Comment (MKhalusova) [reply docs/source/en/tasks/document_question_answering.mdx:55]: I'll check. This is the version that was used in Niels's notebook.

File: docs/source/en/tasks/document_question_answering.mdx (hunk 7 above)
Comment (MKhalusova) [reply docs/source/en/tasks/document_question_answering.mdx:372]: Yes, that's a typo, thank you

File: docs/source/en/tasks/document_question_answering.mdx (hunk 2 above)
Comment (MKhalusova) [reply docs/source/en/tasks/document_question_answering.mdx:56]: I've tried more recent versions with no luck. Perhaps, @NielsRogge can advise on what are the latest versions of torch, torchvision and detectron are supported?

File: docs/source/en/tasks/document_question_answering.mdx [hunk 9]
```diff
@@ -0,0 +1,486 @@
...
+...     for idx, i in enumerate(range(len(words_list))):
+...         if words_list[i] == answer_list[0] and words_list[i : i + len(answer_list)] == answer_list:
+...             matches.append(answer_list)
+...             start_indices.append(idx)
+...             end_indices.append(idx + len(answer_list) - 1)
+...     if matches:
+...         return matches[0], start_indices[0], end_indices[0]
+...     else:
+...         return None, 0, 0
+```
```
Comment (sayakpaul) [inline docs/source/en/tasks/document_question_answering.mdx:234]: I would also briefly describe the function in one / two sentences.

File: docs/source/en/tasks/document_question_answering.mdx [hunk 10]
```diff
@@ -0,0 +1,486 @@
...
+...     answers = examples["answer"]
+...     encoding = tokenizer(questions, words, boxes, max_length=max_length, padding="max_length", truncation=True)
+...     start_positions = []
+...     end_positions = []
+
+...     for i in range(len(questions)):
+...         cls_index = encoding["input_ids"][i].index(tokenizer.cls_token_id)
+...         words_example = [word.lower() for word in words[i]]
+...         answer = answers[i]
+...         print("Answer:", answer)
```
Comment (sayakpaul) [inline docs/source/en/tasks/document_question_answering.mdx:284]: Remove? I guess it's here for debugging.

File: docs/source/en/tasks/document_question_answering.mdx [hunk 11]
```diff
@@ -0,0 +1,486 @@
...
+We'll need to find the position of the answer in the encoded input.
+* `token_type_ids` tells us which tokens are part of the question, and which ones are part of the document's words.
+* `tokenizer.cls_token_id` will help find the special token at the beginning of the input.
+* `word_ids` will help match the answer found in the original `words` to the same answer in the full encoded input and determine
+the start/end position of the answer in the encoded input.
+
+With that in mind, let's create a function to encode a batch of examples in the dataset:
+
+```py
+>>> def encode_dataset(examples, max_length=512):
```
Comment (sayakpaul) [inline docs/source/en/tasks/document_question_answering.mdx:271]: Would be good to add comments above the sub code blocks. A long code block might get hard to follow.

File: docs/source/en/tasks/document_question_answering.mdx [hunk 12]
```diff
@@ -0,0 +1,486 @@
...
+...                     token_start_index += 1
+
+...             for id in word_ids[::-1]:
+...                 if id == word_idx_end:
+...                     end_position = token_end_index
+...                 else:
+...                     token_end_index -= 1
+
+...             print("True answer:", answer)
+...             reconstructed_answer = tokenizer.decode(encoding["input_ids"][i][start_position : end_position + 1])
```
Comment (sayakpaul) [inline docs/source/en/tasks/document_question_answering.mdx:314]: You can also add `skip_special_tokens=True` to not include the special tokens in the decoded output.

File: docs/source/en/tasks/document_question_answering.mdx [hunk 13]
```diff
@@ -0,0 +1,486 @@
...
+...                 else:
+...                     token_start_index += 1
+
+...             for id in word_ids[::-1]:
+...                 if id == word_idx_end:
+...                     end_position = token_end_index
+...                 else:
+...                     token_end_index -= 1
+
+...             print("True answer:", answer)
```
Comment (sayakpaul) [inline docs/source/en/tasks/document_question_answering.mdx:313]: Remove?

File: docs/source/en/tasks/document_question_answering.mdx [hunk 14]
```diff
@@ -0,0 +1,486 @@
...
+...             reconstructed_answer = tokenizer.decode(encoding["input_ids"][i][start_position : end_position + 1])
+...             start_positions.append(start_position)
+...             end_positions.append(end_position)
+...             print("Reconstructed answer:", reconstructed_answer)
+...             print("-----------")
+
```
Comment (sayakpaul) [inline docs/source/en/tasks/document_question_answering.mdx:319]: Remove?

File: docs/source/en/tasks/document_question_answering.mdx [hunk 15]
```diff
@@ -0,0 +1,486 @@
...
+...         else:
+...             print("Answer not found in context")
+...             print("-----------")
```
Comment (sayakpaul) [inline docs/source/en/tasks/document_question_answering.mdx:322]: Remove?

File: docs/source/en/tasks/document_question_answering.mdx [hunk 16]
```diff
@@ -0,0 +1,486 @@
...
+```py
+>>> encoded_train_dataset.features
+{'image': Sequence(feature=Sequence(feature=Sequence(feature=Value(dtype='uint8', id=None), length=-1, id=None), length=-1, id=None), length=-1, id=None),
+ 'input_ids': Sequence(feature=Value(dtype='int32', id=None), length=-1, id=None),
+ 'token_type_ids': Sequence(feature=Value(dtype='int8', id=None), length=-1, id=None),
+ 'attention_mask': Sequence(feature=Value(dtype='int8', id=None), length=-1, id=None),
+ 'bbox': Sequence(feature=Sequence(feature=Value(dtype='int64', id=None), length=-1, id=None), length=-1, id=None),
+ 'start_positions': Value(dtype='int64', id=None),
+ 'end_positions': Value(dtype='int64', id=None)}
+```
```
Comment (sayakpaul) [inline docs/source/en/tasks/document_question_answering.mdx:352]: At this point, I would do one final visualization to ensure the data is prepared in an expected manner:

* Denormalized image
* Decoded questions and answers (for questions, we can use `input_ids`)

File: docs/source/en/tasks/document_question_answering.mdx [hunk 17]
```diff
@@ -0,0 +1,486 @@
...
+
+## Evaluation
+
+Evaluation for document question answering requires a significant amount of postprocessing. To avoid taking up too much
+of your time, this guide skips the evaluation step. The [`~Trainer`] still calculates the evaluation loss during training so
+you're not completely in the dark about your model's performance.
```
Comment (sayakpaul) [inline docs/source/en/tasks/document_question_answering.mdx:358]: If there are any existing resources that the readers could refer to, I would include them here.

File: docs/source/en/tasks/document_question_answering.mdx [hunk 18]
```diff
@@ -0,0 +1,486 @@
...
+>>> from transformers import Trainer
+
+>>> trainer = Trainer(
+...     model=model,
+...     args=training_args,
+...     data_collator=data_collator,
+...     train_dataset=encoded_train_dataset,
+... )
+
+>>> trainer.train()
```
Comment (sayakpaul) [inline docs/source/en/tasks/document_question_answering.mdx:419]: Why are we not using a validation set?

File: docs/source/en/tasks/document_question_answering.mdx [hunk 19]
```diff
@@ -0,0 +1,486 @@
...
+```
+
+To add the final model to 🤗 Hub, save the processor, create a model card, and call `push_to_hub`:
+
+```py
+>>> processor.save_pretrained(repo_id)
+>>> trainer.create_model_card()
+>>> trainer.push_to_hub()
+```
```
Comment (sayakpaul) [inline docs/source/en/tasks/document_question_answering.mdx:428]: Do we explicitly need to save the processor? I thought `trainer.push_to_hub()` would already do it if you specify the `tokenizer` argument to `Trainer`. Cc: @sgugger

**Review by sayakpaul — CHANGES_REQUESTED** [review]
Exceptionally amazing guide, this 🔥

File: docs/source/en/tasks/document_question_answering.mdx (hunk 4 above)
Comment (MKhalusova) [reply docs/source/en/tasks/document_question_answering.mdx]: Fixed in 60cd46474eb9051f3b6c1f1c74e912a0e9ab0f8d

File: docs/source/en/tasks/document_question_answering.mdx (hunk 7 above)
Comment (MKhalusova) [reply docs/source/en/tasks/document_question_answering.mdx]: Fixed in 60cd46474eb9051f3b6c1f1c74e912a0e9ab0f8d

File: docs/source/en/tasks/document_question_answering.mdx [hunk 20]
```diff
@@ -0,0 +1,486 @@
...
+
+>>> processor = LayoutLMv2Processor.from_pretrained(model_checkpoint)
+```
+
+### Preprocessing document images
+
+First, let's prepare the document images for the model with the help of the `image_processor` from the processor.
+By default, image processor resizes the images to 224x224, makes sure they have the correct order of color channels,
+applies OCR with tesseract to get words and normalized bounding boxes. In this tutorial, all of these defaults are exactly what we need.
+Write a function that applies the default image processing to a batch of images and returns the results of OCR.
```
[the remark this answers is not in the document]
Comment (MKhalusova) [reply docs/source/en/tasks/document_question_answering.mdx:181]: Image processor applies OCR with Tesseract but also does other preprocessing.

File: docs/source/en/tasks/document_question_answering.mdx [hunk 21]
```diff
@@ -0,0 +1,486 @@
...
+
+```py
+>>> updated_dataset = updated_dataset.filter(lambda x: len(x["words"]) + len(x["question"].split()) < 512)
+```
+
+At this point let's also remove the OCR features from this dataset. These are a result of OCR for fine-tuning a different
+model. They would still require some processing if we wanted to use them, as they do not match the input requirements
+of the model we use in this guide. Instead, we can use the [`LayoutLMv2Processor`] on the original data for both OCR and
+tokenization. This way we'll get the inputs that match model's expected input. If you want to process images manually,
+check out the [`LayoutLMv2` model documentation](../model_doc/layoutlmv2) to learn what input format the model expects.
```
[the remark this answers is not in the document]
Comment (MKhalusova) [reply docs/source/en/tasks/document_question_answering.mdx:147]: Noted the idea for the guide.

[the remark this answers is not in the document]
Comment (MKhalusova) [reply docs/source/en/tasks/document_question_answering.mdx:147]: The input format information is listed in the overview section of the model doc, I think this link is enough in this context.

File: docs/source/en/tasks/document_question_answering.mdx [hunk 22]
```diff
@@ -0,0 +1,486 @@
...
+
+>>> def get_ocr_words_and_boxes(examples):
+...     images = [image.convert("RGB") for image in examples["image"]]
+...     encoded_inputs = image_processor(images)
+
+...     examples["image"] = encoded_inputs.pixel_values
+...     examples["words"] = encoded_inputs.words
+...     examples["boxes"] = encoded_inputs.boxes
+
+...     return examples
```
[the remark this answers is not in the document]
Comment (MKhalusova) [reply docs/source/en/tasks/document_question_answering.mdx:195]: Mainly because we do not know how they were created in the first place in this dataset and what caveats there may be. I'll add an explanation.

File: docs/source/en/tasks/document_question_answering.mdx [hunk 23]
```diff
@@ -0,0 +1,486 @@
...
+-->
+
+# Document Question Answering
+
+[[open-in-colab]]
+
+Document Question Answering, also referred to as Document Visual Question Answering, is a task that involves providing
+answers to questions posed about document images. The input to these models is typically a combination of an image and
+a question, and the output is an answer expressed in natural language. These models utilize multiple features, including
+text, the positions of words (bounding boxes), and the image itself.
```
Comment (NielsRogge) [inline docs/source/en/tasks/document_question_answering.mdx:20]: It's a bit weird that you start with "these models" => which models?

File: docs/source/en/tasks/document_question_answering.mdx [hunk 24]
```diff
@@ -0,0 +1,486 @@
...
+
+This guide illustrates how to:
+
+- Fine-tune [LayoutLMv2](../model_doc/layoutlmv2) on the [DocVQA dataset](https://huggingface.co/datasets/nielsr/docvqa_1200_examples_donut).
+- Use your fine-tuned model for inference.
+
+<Tip>
+
+The task illustrated in this tutorial is supported by the following model architectures:
```
Comment (NielsRogge) [inline docs/source/en/tasks/document_question_answering.mdx:29]: Technically also Donut, and soon models like Pix2Struct and UDOP can do document question answering (in an entirely different generative way)

File: docs/source/en/tasks/document_question_answering.mdx (hunk 12 above)
Comment (MKhalusova) [reply docs/source/en/tasks/document_question_answering.mdx:314]: I'll remove printing the reconstructed answer altogether as this was mainly for debugging purposes

File: docs/source/en/tasks/document_question_answering.mdx [hunk 25]
```diff
@@ -0,0 +1,486 @@
...
+specific language governing permissions and limitations under the License.
+-->
+
+# Document Question Answering
+
+[[open-in-colab]]
+
+Document Question Answering, also referred to as Document Visual Question Answering, is a task that involves providing
+answers to questions posed about document images. The input to these models is typically a combination of an image and
+a question, and the output is an answer expressed in natural language. These models utilize multiple features, including
```
Comment (NielsRogge) [inline docs/source/en/tasks/document_question_answering.mdx:19]: features => I'd say modalities (as these models are multi-modal)

File: docs/source/en/tasks/document_question_answering.mdx (hunk 18 above)
Comment (MKhalusova) [reply docs/source/en/tasks/document_question_answering.mdx:419]: Because we're not running evaluation.

File: docs/source/en/tasks/document_question_answering.mdx [hunk 26]
```diff
@@ -0,0 +1,486 @@
...
+```py
+>>> model_checkpoint = "microsoft/layoutlmv2-base-uncased"
+>>> batch_size = 4
+```
+
+## Load the data
+
+In this guide we use a small sample of preprocessed DocVQA that you can find on 🤗 Hub. If you'd like to use the full
+DocVQA dataset, you can register and download it on [DocVQA homepage](https://rrc.cvc.uab.es/?ch=17). If you do so, to
+proceed with this guide check out [how to load files into a 🤗 dataset](https://huggingface.co/docs/datasets/loading#local-and-remote-files).
```
Comment (NielsRogge) [inline docs/source/en/tasks/document_question_answering.mdx:90]: We should definitely have a guide on this, cause it's far from straightforward to create a DocVQA dataset as I did

File: docs/source/en/tasks/document_question_answering.mdx [hunk 27]
```diff
@@ -0,0 +1,486 @@
...
+        num_rows: 200
+    })
+})
+```
+
+As you can see, the dataset is split into train and test sets already. Take a look at a random example to familiarize
+yourself with the features.
+
+```py
+>>> dataset["train"][0]
```
Comment (NielsRogge) [inline docs/source/en/tasks/document_question_answering.mdx:113]:
```suggestion
>>> dataset["train"].features
```
If you do `dataset["train"][0]` it will print everything, which might be undesirable

File: docs/source/en/tasks/document_question_answering.mdx [hunk 28]
```diff
@@ -0,0 +1,486 @@
...
+```
+
+Here's what the individual fields represent:
+* `id`: the example's id
+* `image`: a PIL.Image.Image object containing the document image
+* `query`: the question string - natural language asked question, in several languages
+* `answers`: a list of correct answers provided by annotators
```
Comment (NielsRogge) [inline docs/source/en/tasks/document_question_answering.mdx:120]:
```suggestion
* `answers`: a list of correct answers provided by human annotators
```

File: docs/source/en/tasks/document_question_answering.mdx (hunk 18 above)
Comment (sayakpaul) [reply docs/source/en/tasks/document_question_answering.mdx:419]: Sure but having a validation set to see if the validation loss is going down would be quite helpful IMO.

File: docs/source/en/tasks/document_question_answering.mdx [hunk 29]
```diff
@@ -0,0 +1,486 @@
...
+
+```py
+>>> updated_dataset = dataset.map(lambda example: {"question": example["query"]["en"]}, remove_columns=["query"])
+>>> updated_dataset = updated_dataset.map(
+...     lambda example: {"answer": example["answers"][0]}, remove_columns=["answer", "answers"]
+... )
+```
+
+Note that the LayoutLMv2 checkpoint that we use in this guide has been trained with `max_position_embeddings = 512` (you can
+find this information in the [checkpoint's `config.json` file](https://huggingface.co/microsoft/layoutlmv2-base-uncased/blob/main/config.json)).
```
Comment (NielsRogge) [inline docs/source/en/tasks/document_question_answering.mdx:134]:
```suggestion
find this information in the [checkpoint's `config.json` file](https://huggingface.co/microsoft/layoutlmv2-base-uncased/blob/main/config.json#L18)).
```

File: docs/source/en/tasks/document_question_answering.mdx [hunk 30]
```diff
@@ -0,0 +1,486 @@
...
+We can truncate the examples but to avoid the situation where the answer might be at the end of a large document and end up truncated,
+let's remove the few examples where the embedding is likely to end up longer than 512. If most of the documents in your dataset are long, you
+may need to choose a different pretrained model checkpoint.
```
[the remark this answers is not in the document]
Comment (NielsRogge) [reply docs/source/en/tasks/document_question_answering.mdx:137]: Hmm all pretrained checkpoints have a max token length of 512.

If most of the documents are long, it's recommended to use a sliding window approach (see the official QA notebook)

File: docs/source/en/tasks/document_question_answering.mdx [hunk 31]
```diff
@@ -0,0 +1,486 @@
...
+
+### Preprocessing document images
+
+First, let's prepare the document images for the model with the help of the `image_processor` from the processor.
+By default, image processor resizes the images to 224x224, makes sure they have the correct order of color channels,
+applies OCR with tesseract to get words and normalized bounding boxes. In this tutorial, all of these defaults are exactly what we need.
+Write a function that applies the default image processing to a batch of images and returns the results of OCR.
+
+```py
+>>> image_processor = processor.image_processor
```
Comment (NielsRogge) [inline docs/source/en/tasks/document_question_answering.mdx:184]: It's actually recommended to directly use the processor and provide `images` as keyword argument

File: docs/source/en/tasks/document_question_answering.mdx (hunk 17 above)
Comment (NielsRogge) [reply docs/source/en/tasks/document_question_answering.mdx:358]: Extractive question answering is typically evaluated using F1/exact match

File: docs/source/en/tasks/document_question_answering.mdx (hunk 19 above)
Comment (sgugger) [reply docs/source/en/tasks/document_question_answering.mdx:428]: Yes if you pass `tokenizer=processor` you won't need to save the processor independently.
(We really need to rename that argument in the Trainer :sweat_smile: )

Comment (MKhalusova) [reply docs/source/en/tasks/document_question_answering.mdx:428]: You're right, we can pass the processor as a tokenizer to the trainer, and it'll be pushed. I'll fix this.

File: docs/source/en/tasks/document_question_answering.mdx (hunk 24 above)
Comment (MKhalusova) [reply docs/source/en/tasks/document_question_answering.mdx:29]: The tip is generated from model mapping MODEL_FOR_DOCUMENT_QUESTION_ANSWERING_MAPPING_NAMES. Is there more than one mapping for this task?

File: docs/source/en/tasks/document_question_answering.mdx (hunk 26 above)
Comment (MKhalusova) [reply docs/source/en/tasks/document_question_answering.mdx:82]: Sounds like a good followup PR :) Would be awesome if you could do it, @sayakpaul !

File: docs/source/en/tasks/document_question_answering.mdx (hunk 30 above)
[the remark this answers is not in the document]
Comment (MKhalusova) [reply docs/source/en/tasks/document_question_answering.mdx:130]: Did not know that, thank you! I'll update the doc.

File: docs/source/en/tasks/document_question_answering.mdx (hunk 2 above)
Comment (MKhalusova) [reply docs/source/en/tasks/document_question_answering.mdx]: Fixed

File: docs/source/en/tasks/document_question_answering.mdx (hunk 23 above)
Comment (MKhalusova) [reply docs/source/en/tasks/document_question_answering.mdx:20]: It is described the same way as in the Task description.  https://huggingface.co/tasks/document-question-answering

File: docs/source/en/tasks/document_question_answering.mdx [hunk 32]
```diff
@@ -0,0 +1,486 @@
...
+```
+
+Finally, the data exploration won't be complete if we don't peek at an image example.
+
+```py
+>>> updated_dataset["train"][11]["image"]
+```
+
+<div class="flex justify-center">
+     <img src="https://imgur.com/VfOeFjc" alt="DocVQA Image Example"/>
```
[the remark this answers is not in the document]
Comment (MKhalusova) [reply docs/source/en/tasks/document_question_answering.mdx:157]: I don't have permissions to merge :( https://huggingface.co/datasets/huggingface/documentation-images/discussions/43

File: docs/source/en/tasks/document_question_answering.mdx (hunk 9 above)
Comment (MKhalusova) [reply docs/source/en/tasks/document_question_answering.mdx:236]: Fixed

File: docs/source/en/tasks/document_question_answering.mdx (hunk 30 above)
[the remark this answers is not in the document]
Comment (MKhalusova) [reply docs/source/en/tasks/document_question_answering.mdx]: Updated.

File: docs/source/en/tasks/document_question_answering.mdx (hunk 18 above)
Comment (MKhalusova) [reply docs/source/en/tasks/document_question_answering.mdx]: Fixed

File: docs/source/en/tasks/document_question_answering.mdx (hunk 16 above)
Comment (MKhalusova) [reply docs/source/en/tasks/document_question_answering.mdx:359]: I am not sure if visualization of the result adds value in this example. The images are simply resized, and most of the processing is about boxes and text.

File: docs/source/en/tasks/document_question_answering.mdx (hunk 32 above)
[the remark this answers is not in the document]
Comment (MKhalusova) [reply docs/source/en/tasks/document_question_answering.mdx]: Thanks for merging!

**Comment by sayakpaul** [comment]
@MKhalusova is it okay if I reviewed the latest changes on Monday? I am currently on travel.

**Comment by MKhalusova** [comment]
> @MKhalusova is it okay if I reviewed the latest changes on Monday? I am currently on travel.

Absolutely!

**Review by sayakpaul — APPROVED** [review]
Awesome work!
